### DISCUSSION TITLE
Windows如何实现在视频中渲染pag动态图呢

### DISCUSSION THREAD

**Discussion by jackTJT** [Tencent/libpag#1866] [General] [answered]
我想利用libpag和ffmpeg在Windows上实现，在一个视频中渲染一个pag动图，请问该如何做呢，是需要先把视频读取成帧然后再将pagde帧插入到其中吗

**Comment by kevingpqi123** [comment, marked as answer]
是的，视频中添加特效都是按帧进行处理，获取到每帧的渲染结果，然后做渲染结果合并，然后再进行视频编码
pag 专注于动效渲染，视频编辑是另外一个技术领域，不在我们技术支持的范围之内，需要开发者自己处理

**Reply by jackTJT** [reply]
请问我们在视频解码后，提供图像帧后(例如rgb格式)，然后同时pag库读取pag文件，得到渲染素材后，能够用pag库进行一个渲染叠加吗，素材和图像帧进行叠加

**Reply by kevingpqi123** [reply]
如果你这边视频的内容作为占位图传入到 pag 中，那么 flush 之后会输出整体效果，如果 pag 作为贴纸 放在视频中的某个区域，需要自己处理叠加，这里的叠加就是在一个 Surface 上绘制两次，先绘制视频画面，再绘制 pag 效果，没你想象的那么复杂
